rook/scripts/funcs.sh

```diff
@@ -0,0 +1,407 @@
+#!/bin/bash
+LOGGING=${LOGGING:-false}
+PIPELINE=${PIPELINE:-false}
+here=${here:-null}
+if [ -z "$PIPELINE" ]
+then
```

Comment:
Let's go with this format throughout the script
```suggestion
if [ -z "$PIPELINE" ]; then
```